ergo-lib/src/wallet/tx_builder.rs

```diff
@@ -131,15 +162,26 @@ impl<S: ErgoBoxAssets + ErgoBoxId + Clone> TxBuilder<S> {
             .iter()
             .filter(|b| b.value >= self.min_change_value)
             .map(|b| {
-                ErgoBoxCandidateBuilder::new(
+                let mut candidate = ErgoBoxCandidateBuilder::new(
                     b.value,
                     change_address_ergo_tree.clone(),
                     self.current_height,
-                )
-                .build()
+                );
+                for token in &b.tokens() {
```

Comment:
Note: this is a change in the behavior for how TxBuilder handles change. It used not include tokens inside the change address, which means that transactions that didn't explicitly send input tokens to an output failed.

Reply:
Oops, my bad. Forgot to put the tokens. Good catch! Thanks for the fix!